src/AdvancedHMC.jl

```diff
@@ -12,6 +12,8 @@ using ArgCheck: @argcheck
 
 import StatsBase: sample
 
+function renew end  # this is needed as `AbstractMetric` are in `Adaptation`
```

Comment:
Maybe replace `renew` with `Parameters.reconstruct`?  See e.g. 

https://mauro3.github.io/Parameters.jl/latest/api.html#Parameters.reconstruct-Union{Tuple{T},%20Tuple{T,Any}}%20where%20T 

We already make use of `Parameters.unpack` in several places of `AHMC`.

Reply:
Looks like a good idea. I will give a try. So maybe it's better for me to implement `renew` using `reconstruct`, instead of replacing it? Otherwise I will lose the simplcity here: https://github.com/TuringLang/AdvancedHMC.jl/pull/81/files#diff-f48defe9803ddea4f759bce857b70de0R514.